artcommon/artcommonlib/github_auth.py

```
@@ -0,0 +1,169 @@
+    :raises ValueError: If no installations are found for the app
+    """
+    app_auth = Auth.AppAuth(app_id, private_key)
+
+    if installation_id is None:
+        gi = GithubIntegration(auth=app_auth)
+        installations = gi.get_installations()
+        if not installations:
+            raise ValueError(
+                f"No installations found for GitHub App {app_id}. "
+                "Ensure the app is installed on at least one organization or repository."
+            )
```

Ruff (0.15.5)

[warning] Avoid specifying long messages outside the exception class

(TRY003)